control-plane/grpc/protos/pool.proto

```diff
@@ -0,0 +1,133 @@
+syntax = "proto3";
+
+import "common.proto";
+import "google/protobuf/empty.proto";
+import "google/protobuf/wrappers.proto";
+
+package grpc;
+
+// A Mayastor Storage Pool
+// It may have a spec which is the specification provided by the creator
+// It may have a state if such state is retrieved from a mayastor storage node
+message Pool {
+  // Desired specification of the pool and metadata
+  optional PoolDefinition definition = 1;
+  // Runtime state of the pool.
+  optional PoolState state = 2;
+}
+
+// Multiple pools
+message Pools {
+  repeated Pool pools = 1;
+}
+
+// PoolLabel is the type for the labels
+message PoolLabel {
+  map<string, string> labels = 1;
+}
+
+// Desired spec and the control plane related data
+message PoolDefinition {
+  // Client specification of the pool
+  PoolSpec spec = 1;
+  // Control plane related data for the pool
+  Metadata metadata = 2;
+}
+
+message Metadata {
```

Comment:
I think we should include a comment for this message.

Reply:
Added comment.